include/matx/operators/scalar_ops.h

```diff
@@ -170,8 +170,38 @@ template <typename T> struct SqrtF {
   }
 };
 
-template <typename T> using SqrtOp = UnOp<T, SqrtF<T>>;
+template <typename T>
+static __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ auto _internal_csqrt(T v1)
+{
+  if constexpr (is_cuda_complex_v<T>) 
```

Comment:
Do we want to do this or return an error if they pass a complex to this since it doesn't follow C++ libs?

Reply:
It currently fails to compile, because it can't complete the type; for example if it's an int,  `complex<int>` can't be implemented so it errors. one sec, I'll post the exact error message. I thought it was relatively self-explanatory, but I can add better checks if we want

Reply:
error generates for trying to feed in an int:

````
/home/tallen/scratch/playground/matxExample/build/_deps/cccl-src/libcudacxx/include/cuda/std/detail/libcxx/include/complex(1382): error: no instance of function template "cuda::std::__4::polar" matches the argument list
            argument types are: (double, int)
      return cuda::std::__4::polar(cuda::std::__4::sqrt(cuda::std::__4::abs(__x)), cuda::std::__4::arg(__x) / _Tp(2));
             ^
          detected during instantiation of "cuda::std::__4::complex<_Tp> cuda::std::__4::sqrt(const cuda::std::__4::complex<_Tp> &) [with _Tp=int]" at line 182 of /home/tallen/scratch/playground/matxExample/MatX/include/matx/operators/scalar_ops.h

1 error detected in the compilation of "/home/tallen/scratch/playground/matxExample/complexSetVals.cu".
make[2]: *** [CMakeFiles/complexSetVals.dir/build.make:76: CMakeFiles/complexSetVals.dir/complexSetVals.cu.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:201: CMakeFiles/complexSetVals.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

Reply:
so it does not currently point back to the line where the int is being used, but it does complain about int in a complex... so probably could be more helpful

Reply:
I thought this function is only for scalars of floating point type. Can this just assert it's a floating point type, and if so, just call the `cuda::std` version and that's it?

Reply:
this should take in a float type (non complex) and return a complex type of the same float type as the input. Ie:
`float` -> `complex<float>`
`double` -> `complex<double>`

I _thought_ the C++ standard also supported `complex<float>` -> `complex<float>` for it's csqrt, but it looks like the problem is that the `csqrt `function is a c function (C99), and does not support the `std::complex`, so it does not support complex inputs.

so I think the valid types is actually just this:

`
template <typename T>
static __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ auto _internal_csqrt(T v1)
std::enable_if_t<std::is_floating_point_v<T>, bool> = true>
{
  return sqrt(static_cast<cuda::std::complex<T>>(v1));
}
`

if you think that makes sense I'll update the code to the above.

Reply:
we could easily make ours support `complex<float>` as well (it would just call the normal sqrt), but that would go beyond the standard C++ support pattern.

Reply:
I would remove the `enable_if` and instead make it a `static_assert` with a message telling the user it's not supported. otherwise they get a bad template error they have to figure out.

Reply:
alrighty, updated per our discussion. Tested it all works as expected.
Works with float/double, errors on any other type (example error below for using int)

```
Consolidate compiler generated dependencies of target complexSetVals
[ 20%] Built target cuBlasExample
[ 50%] Building CUDA object CMakeFiles/nvtx_autoGen.dir/nvtx_autoGen.cu.o
[ 50%] Building CUDA object CMakeFiles/vectorMult.dir/vectorMult.cu.o
[ 50%] Building CUDA object CMakeFiles/matxExample.dir/test.cu.o
[ 60%] Building CUDA object CMakeFiles/complexSetVals.dir/complexSetVals.cu.o
/home/tallen/scratch/playground/matxExample/MatX/include/matx/operators/scalar_ops.h(176): error: static assertion failed with "csqrt() only supports non-complex floating point inputs"
    static_assert(std::is_floating_point_v<T>, "csqrt() only supports non-complex floating point inputs");
    ^
          detected during:
            instantiation of "auto matx::detail::_internal_csqrt(T) [with T=int]" at line 184
            instantiation of "auto matx::detail::CSqrtF<T>::op(T) [with T=int]" at line 102
            instantiation of "auto matx::detail::UnOp<T1, F>::op(const T1 &) [with T1=int, F=matx::detail::CSqrtF<int>]" at line 106
            instantiation of class "matx::detail::UnOp<T1, F> [with T1=int, F=matx::detail::CSqrtF<int>]" at line 365 of /home/tallen/scratch/playground/matxExample/MatX/include/matx/operators/unary_operators.h
            instantiation of "auto matx::csqrt(I1) [with I1=matx::tensor_t<int, 1, matx::basic_storage<matx::raw_pointer_buffer<int, matx::matx_allocator<int>>>, matx::tensor_desc_t<std::array<matx::index_t, 1UL>, std::array<matx::index_t, 1UL>, 1>>, <unnamed>=void]" at line 53 of /home/tallen/scratch/playground/matxExample/complexSetVals.cu

1 error detected in the compilation of "/home/tallen/scratch/playground/matxExample/complexSetVals.cu".
make[2]: *** [CMakeFiles/complexSetVals.dir/build.make:76: CMakeFiles/complexSetVals.dir/complexSetVals.cu.o] Error 2
```